pkg/util/homedir/homedir.go

```diff
@@ -23,7 +23,7 @@ import (
 
 // HomeDir returns the home directory for the current user
 func HomeDir() string {
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS == "windows"; len(os.Getenv("HOME")) <= 0) {
```

Comment:
You will probably need a migration [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader.go#L56-L66) (Windows only) to make sure that users get their configs migrated correctly if they are relying on the current approach.